packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1012,7 +1012,8 @@ function useMutableSource<Source, Snapshot>(
       const latestSetSnapshot = refs.setSnapshot;
 
       try {
-        latestSetSnapshot(latestGetSnapshot(source._source));
+        const value = latestGetSnapshot(source._source);
+        latestSetSnapshot(() => value);
```

Comment:
This is an issue that I chatted about with Daishi a few weeks back, which has this obvious fix but... I had hoped to avoid the [almost always] unnecessary function wrapper. Still don't know how I feel about it.

Reply:
Andrew's comment just loaded in for me after writing mine.

I agree with the concern he mentions about opening up extra problem cases if the function itself closes over mutable values.

> As an alternative, could we say we don't support functions as snapshots and warn in dev if we detect one?

I tend to like this better.